src/QEther.cpp

```diff
@@ -24,32 +24,75 @@
 
 using namespace dev::mix;
 
+void QEther::manageException() const
+{
+	try
+	{
+		throw;
+	}
+	catch (boost::exception const& _e)
+	{
+		std::cerr << boost::diagnostic_information(_e);
+	}
+	catch (std::exception const& _e)
+	{
+		std::cerr << _e.what();
+	}
+	catch (...)
+	{
+		std::cerr << boost::current_exception_diagnostic_information();
+	}
+}
+
 QString QEther::format() const
 {
-	return QString::fromStdString(dev::eth::formatBalance(boost::get<dev::u256>(toWei()->internalValue())));
+	try
+	{
+		return QString::fromStdString(dev::eth::formatBalance(boost::get<dev::u256>(toWei()->internalValue())));
+	}
+	catch (...)
+	{
+		manageException();
+		return QString();
+	}
 }
 
 QBigInt* QEther::toWei() const
 {
-	QMetaEnum units = staticMetaObject.enumerator(staticMetaObject.indexOfEnumerator("EtherUnit"));
-	const char* key = units.valueToKey(m_currentUnit);
-	for (std::pair<dev::u256, std::string> rawUnit: dev::eth::units())
+	try
+	{
+		QMetaEnum units = staticMetaObject.enumerator(staticMetaObject.indexOfEnumerator("EtherUnit"));
+		const char* key = units.valueToKey(m_currentUnit);
+		for (std::pair<dev::u256, std::string> rawUnit: dev::eth::units())
+		{
+			if (QString::fromStdString(rawUnit.second).toLower() == QString(key).toLower())
+				return multiply(new QBigInt(rawUnit.first));
+		}
+		return new QBigInt(dev::u256(0));
+	}
+	catch (...)
 	{
-		if (QString::fromStdString(rawUnit.second).toLower() == QString(key).toLower())
-			return multiply(new QBigInt(rawUnit.first));
+		manageException();
+		return nullptr;
```

Comment:
not sure if returning 0 will change something. Returning either 0 or nullptr gives a null object in JavaScript.